WordPress/Classes/ViewRelated/System/Sidebar/SidebarViewController.swift

```diff
@@ -162,21 +162,31 @@ private struct SidebarProfileContainerView: View {
     }
 }
 
-private extension BlogListViewModel {
-    /// Returns the most recent sites and mixes and mixes in the rest of the sites
+extension BlogListViewModel {
+    /// Returns the most recent sites, the site for given blog id, and mixes in the rest of the sites
     /// until the display limimt is reached.
-    var topSites: [BlogListSiteViewModel] {
-        var topSites = recentSites.prefix(SidebarView.displayedSiteLimit)
+    func topSites(limit: Int, containing blogId: TaggedManagedObjectID<Blog>?) -> [BlogListSiteViewModel] {
+        var topSites = recentSites.prefix(limit)
         var encounteredIDs = Set(topSites.map(\.id))
         for site in allSites where !encounteredIDs.contains(site.id) {
-            if topSites.count >= SidebarView.displayedSiteLimit {
+            if topSites.count >= limit {
                 break
             }
             encounteredIDs.insert(site.id)
             topSites.append(site)
         }
-        return Array(topSites).sorted { $0.title.localizedCaseInsensitiveCompare($1.title) == .orderedAscending
+
+        if let blogId, !encounteredIDs.contains(blogId), let site = allSites.first(where: { $0.id == blogId }) {
```

Comment:
I'm not sure it should be necessary because the list displays the most recent sites. When you select a site, the system has to add it to `RecentSitesService` and the list needs to be reload. If the list isn't reloading or the site isn't added to the recent sites, that would be the root cause of the issue. I'd suggest addressing the root cause.

It might be worth keeping this as defensive code, but I'm not sure. Wdyt?

Reply:
The`RecentSitesService` only stores a list of recently used sites. It doesn't know which one is the most recent one though.

Reply:
The most recent one should be the first one of the `TheRecentSitesService` list by the time the sidebar selection finishes. The sites are also sorted alphabetically in the sidebar to ensure they don't jump around when recent sites list changes, unless you are selecting something from "All Sites" that wasn't previously on the list. I might be missing some scenario?

Reply:
I don't think that's how it works. It only makes sure the site is in the recent sites list, without re-ordering the list.

https://github.com/wordpress-mobile/WordPress-iOS/blob/65a16fc3fd8487d4ac3623573bf57fd59222d2c0/WordPress/Classes/Services/RecentSitesService.swift#L70-L78

Also, even if `RecentSitesService` keeps most recent site at the first element, those sites will still be sorted in alphabetical order here in the `topSites` function.